cppapi/server/fwdattrdesc.cpp

```diff
@@ -180,7 +181,13 @@ bool FwdAttr::validate_fwd_att(vector<AttrProperty> &prop_list,const string &dev
 	}
 	catch (...) {}
 
-	if (root_att_db_defined == true)
+	//check if full_root_att is already set
+	if (full_root_att.size()!=0 && full_root_att.compare(RootAttNotDef)!=0)
```

Comment:
Sorry, but I still don't get it. This is the original code before any modification:

```c++
if (root_att_db_defined == true)
    full_root_att = root_att_db;
else
    full_root_att = RootAttNotDef;
```

And this is what you are suggesting (correct me if I am wrong):
```c++
if(!is_full_root_att_set && root_att_db_defined)
    full_root_att = root_att_db;
else
    full_root_att = RootAttNotDef;
```
In this case if is_full_root_att_set is true then we would enter the else clause and hence we would modify the full_root_att variable